app/models/solidus_friendly_promotions/promotion_code.rb

```diff
@@ -2,7 +2,7 @@
 
 module SolidusFriendlyPromotions
   class PromotionCode < Spree::Base
-    belongs_to :promotion, inverse_of: :codes
+    belongs_to :promotion, class_name: "SolidusFriendlyPromotions::Promotion", inverse_of: :codes
     belongs_to :promotion_code_batch, inverse_of: :promotion_codes, optional: true
```

Comment:
Why is this only on the promotion and not also on the promotion_code_batch?

Reply:
It's not necessary in the end, rebased it out.